In [1]:
import pandas as pd

Com o objetivo de utilizar dados diferentes dos utilizados na aula, irei utilizar os dados da empresa Magazine Luiza, disponibilizados na mesma plataforma do Yahoo Finance, o arquivo _.csv_ foi hospedado no Git Hub.

In [5]:
df = pd.read_csv("https://raw.githubusercontent.com/EricRaniere/DataScienceNaPratica/main/IntPandasMGLU3SA.csv")

O comando _pd.read__ permite a leitura de diversos tipos de arquivos.

In [6]:
type(df)

pandas.core.frame.DataFrame

O comando type, permite identificar qual o tipo de objeto.

In [7]:
type(df['Date'])

pandas.core.series.Series

In [8]:
df.shape

(246, 7)

O comando _.shape_ permite entender qual o formato do DataFrame(df) selecionado. Nesse caso o _df_ possui 246 linhas e 7 colunas.

In [9]:
df.head()

Date       Open       High  ...      Close  Adj Close      Volume
0  2020-08-06  21.000000  21.547501  ...  21.477501  21.439608  44079200.0
1  2020-08-07  21.299999  21.875000  ...  21.412500  21.374723  46478800.0
2  2020-08-10  21.545000  21.575001  ...  20.482500  20.446363  61277600.0
3  2020-08-11  20.674999  20.747499  ...  20.247499  20.211777  42669200.0
4  2020-08-12  20.327499  20.532499  ...  20.014999  19.979687  44144400.0

[5 rows x 7 columns]

O comando _.head_ por padrão exibe a visualização das 5 primeiras linhas. Caso seja necessário uma visualização maior deve se colocar um número entre os _()_

In [10]:
df.tail()

Date       Open       High  ...      Close  Adj Close      Volume
241  2021-07-30  21.440001  21.700001  ...  20.600000  20.600000  30461200.0
242  2021-08-02  20.910000  21.070000  ...  20.629999  20.629999  23874900.0
243  2021-08-03  20.600000  20.650000  ...  20.430000  20.430000  18984200.0
244  2021-08-04  20.290001  20.320000  ...  19.959999  19.959999  27526900.0
245  2021-08-05   0.000000   0.000000  ...  20.459999  20.459999         0.0

[5 rows x 7 columns]

O comando _.tail_ por padrão exibe a visualização das 5 últimas linhas. Caso seja necessário uma visualização maior deve se colocar um número entre os _()_

In [11]:
df.columns

Index(['Date', 'Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume'], dtype='object')

Com o comando _.columns_ é possível extrair apenas os nomes das colunas do _df_

In [12]:
df.dtypes

Date          object
Open         float64
High         float64
Low          float64
Close        float64
Adj Close    float64
Volume       float64
dtype: object

Com o comando _.dtypes_ é possível conhecer quais as váriaveis compôe o _df_

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 246 entries, 0 to 245
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       246 non-null    object 
 1   Open       245 non-null    float64
 2   High       245 non-null    float64
 3   Low        245 non-null    float64
 4   Close      245 non-null    float64
 5   Adj Close  245 non-null    float64
 6   Volume     245 non-null    float64
dtypes: float64(6), object(1)
memory usage: 13.6+ KB


Outra forma de verificar o formato das váriaveis é o _.info()_. Um diferencial desse comando é a contagem de células não nulas, _non-null_ e o uso de memória do _df_

In [14]:
# selecionar a variável (coluna) "High" da nossa variável df
df['High']

0      21.547501
1      21.875000
2      21.575001
3      20.747499
4      20.532499
         ...    
241    21.700001
242    21.070000
243    20.650000
244    20.320000
245     0.000000
Name: High, Length: 246, dtype: float64

In [15]:
# outra forma de utilização seria com df.High
df.High

0      21.547501
1      21.875000
2      21.575001
3      20.747499
4      20.532499
         ...    
241    21.700001
242    21.070000
243    20.650000
244    20.320000
245     0.000000
Name: High, Length: 246, dtype: float64

Seguindo esse forma podemos ter diversas visualizações estatísticas interessantes.

In [31]:
print('média', df.High.mean())
print('mediana', df.High.median())
print('máximo', df.High.max())
print('mínimo', df.High.min())

média 23.1125101632653
mediana 22.945
máximo 28.309998999999998
minimo 0.0


Uma utilização muito interessante do Pandas é a análise de dados de datas. 

In [32]:
df.Date

0      2020-08-06
1      2020-08-07
2      2020-08-10
3      2020-08-11
4      2020-08-12
          ...    
241    2021-07-30
242    2021-08-02
243    2021-08-03
244    2021-08-04
245    2021-08-05
Name: Date, Length: 246, dtype: object

Como vimos antes, há a necessidade de converter o formato da coluna de _object_ para _datetime_

In [33]:
df.Date = pd.to_datetime(df.Date, format='%Y-%m-%d')

In [35]:
df.Date

0     2020-08-06
1     2020-08-07
2     2020-08-10
3     2020-08-11
4     2020-08-12
         ...    
241   2021-07-30
242   2021-08-02
243   2021-08-03
244   2021-08-04
245   2021-08-05
Name: Date, Length: 246, dtype: datetime64[ns]

Depois da conversão para o formato _datetime_ é possível realizar análises específicas dessa categoria de variável.

In [41]:
# visualização dos anos somente
df.Date.dt.year

0      2020
1      2020
2      2020
3      2020
4      2020
       ... 
241    2021
242    2021
243    2021
244    2021
245    2021
Name: Date, Length: 246, dtype: int64

In [42]:
# visualização dos meses somente
df.Date.dt.month

0      8
1      8
2      8
3      8
4      8
      ..
241    7
242    8
243    8
244    8
245    8
Name: Date, Length: 246, dtype: int64

In [43]:
# visualização dos dias somente
df.Date.dt.day

0       6
1       7
2      10
3      11
4      12
       ..
241    30
242     2
243     3
244     4
245     5
Name: Date, Length: 246, dtype: int64